src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -1963,25 +2006,21 @@ def CommissionOnNetwork(self, nodeId: int, setupPinCode: int,
         '''
         self.CheckIsActive()
 
-        # IP connection will run through full commissioning, so we need to wait
-        # for the commissioning complete event, not just any callback.
-        self.state = DCState.COMMISSIONING
-
         # Convert numerical filters to string for passing down to binding.
         if isinstance(filter, int):
             filter = str(filter)
 
-        self._ChipStack.commissioningCompleteEvent.clear()
+        self._commissioning_complete_future = concurrent.futures.Future()
+        try:
+            self._enablePairingCompeleteCallback(True)
+            self._ChipStack.Call(
+                lambda: self._dmLib.pychip_DeviceController_OnNetworkCommission(
+                    self.devCtrl, self.pairingDelegate, nodeId, setupPinCode, int(filterType), str(filter).encode("utf-8") + b"\x00" if filter is not None else None, discoveryTimeoutMsec)
+            ).raise_on_error()
```

Comment:
Function was not raising before.

See other changed functions as well.

Reply:
Similar to the first instance, it was through `CallAsyncWithCompleteCallback`. Marking as resolved as well.